prov/usnic/src/usdf_fabric.c

```diff
@@ -341,6 +341,82 @@ static int validate_modebits(uint32_t version, const struct fi_info *hints,
 	return FI_SUCCESS;
 }
 
+static int usdf_alloc_fid_nic(struct fi_info *fi,
+			struct usd_device_attrs *dap)
+{
+	int ret;
+	struct fid_nic *nic = NULL;
+	struct fi_device_attr *da = NULL;
+	struct fi_bus_attr *ba = NULL;
+	struct fi_link_attr *la = NULL;
+
+	nic = ofi_nic_dup(NULL);
+	if (!nic)
+		goto nomem;
```

Comment:
The `nomem` label will dereference `nic`, so this will likely cause a crash in the error path.